src/vllm_tgis_adapter/__main__.py

```diff
@@ -328,41 +61,30 @@ async def run_http_server(
         usage_context=UsageContext.OPENAI_API_SERVER,
     )
 
-    event_loop: asyncio.AbstractEventLoop | None
-    try:
-        event_loop = asyncio.get_running_loop()
-    except RuntimeError:
-        event_loop = None
-
-    if event_loop is not None and event_loop.is_running():
-        # If the current is instanced by Ray Serve,
-        # there is already a running event loop
-        model_config = event_loop.run_until_complete(engine.get_model_config())
-    else:
-        # When using single vLLM without engine_use_ray
-        model_config = asyncio.run(engine.get_model_config())
+    loop = asyncio.new_event_loop()
+    tasks = loop.run_until_complete(start_servers(args))
 
-    if event_loop is None:
-        event_loop = asyncio.new_event_loop()
+    def signal_handler() -> None:
+        # prevents the uvicorn signal handler to exit early
+        for task in tasks:
+            task.cancel()
 
-    async def run() -> None:
+    async def override_signal_handler() -> None:
         loop = asyncio.get_running_loop()
-
-        http_server_task = loop.create_task(run_http_server(engine, args, model_config))
-        grpc_server_task = loop.create_task(
-            run_grpc_server(
-                engine, args, disable_log_stats=engine_args.disable_log_stats
```

Comment:
Note that I've gotten rid of the `disable_log_stats` argument, as the `_force_log` coroutine is scheduled as part of the `lifespan` attribute of the fastapi server.